File: Backend/src/main/java/it/unicam/cs/ids/smartchalet/Model/AuthCredential.java
package it.unicam.cs.ids.smartchalet.Model;

import lombok.*;
import org.springframework.data.annotation.Id;
//import org.springframework.security.core.GrantedAuthority;
//import org.springframework.security.core.authority.SimpleGrantedAuthority;
//import org.springframework.security.core.userdetails.UserDetails;
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class AuthCredential  {

    @EqualsAndHashCode.Include
    @Id
    private String mail;
    private String password;
    private List<Role> roles;

    public enum Role {
        CLIENT, BAR_STAFF, ENTERTAINER, NOT_COMPLETED, MANAGER, RECEPTION_STAFF
    }

    //@Override
    //public Collection<? extends GrantedAuthority> getAuthorities() {
        //List<GrantedAuthority> auth = new ArrayList<GrantedAuthority>();
        //for (Role role : roles)
       //     auth.add(new SimpleGrantedAuthority(role.name()));
     //   return auth;
   // }


    public String getPassword() {
        return password;
    }


    public String getUsername() {
        return mail;
    }


    public boolean isAccountNonExpired() {
        return true;
    }


    public boolean isAccountNonLocked() {
        return true;
    }


    public boolean isCredentialsNonExpired() {
        return true;
    }


    public boolean isEnabled()  {
        return true;
    }
}

File: Frontend/smart_chalet/lib/Model/auth_credential.dart
// ignore_for_file: constant_identifier_names

class AuthCredential {
  String mail;
  String password;
  List<Role> _roles = [];

  AuthCredential({
    required this.mail,
    required this.password,
  });

  AuthCredential.complete(
    this.mail,
    this.password,
    this._roles,
  );

  factory AuthCredential.fromJson(Map<String, dynamic> json) {
    return AuthCredential(mail: json["mail"], password: json["password"]);
  }

  String get getMail => mail;

  set setMail(String mail) => this.mail = mail;

  get getPassword => password;

  set setPassword(password) => this.password = password;

  get getRoles => _roles;

  set setRoles(value) => _roles = value;

  List<String> getRolesToString() {
    List<String> roles = [];
    for (Role item in _roles) {
      roles.add(item.toString().substring(item.toString().indexOf('.') + 1));
    }
    return roles;
  }
}

enum Role {
  CLIENT,
  BARSTAFF,
  ENTERTAINER,
  MANAGER,
  RECEPTIONSTAFF,
}
